app/models/payment.rb

```diff
@@ -0,0 +1,14 @@
+class Payment < ActiveRecord::Base
+  belongs_to :account, inverse_of: :payments
+  belongs_to :statement, inverse_of: :payments
+  belongs_to :paid_by, class_name: "User"
+
+  # Add additional sources in an engine with Payment.valid_sources << :new_source
+  def self.valid_sources
+    @@valid_sources ||= [:check]
+  end
+
+  validates :source, presence: true, inclusion: { in: valid_sources }
+  validates :account, :amount, presence: true
+
+end
```

Comment:
I hadn't really thought of the case where the check is invalid or bounces or whatever. I would probably assume we would add it when the check is deposited. I mainly used checks as an example; this is mainly going to be used imminently for credit card payments, which we will only record on capture success. 

We're going to ignore chargebacks and refunds for now. We may give Payments a status at some later time if we expand this feature.